libraries/AP_Eigen/AP_Eigen.h

```diff
@@ -0,0 +1,35 @@
+#ifndef AP_EIGEN_H
+#define AP_EIGEN_H
+
+/*
+ * Eigen interface extensions
+ */
+#define EIGEN_MATRIXBASE_PLUGIN <AP_Eigen/AP_BaseAddons.h>
+#define EIGEN_MATRIX_PLUGIN     <AP_Eigen/AP_MatrixAddons.h>
+
+/*
+ * The PX4 stack creates a lot of drama if just using C99 math features :D
+ */
+#if CONFIG_HAL_BOARD == HAL_BOARD_PX4 && CONFIG_HAL_BOARD != HAL_BOARD_LINUX // maybe more?!
+    #pragma once
+    #pragma GCC diagnostic push
+        #define RAND_MAX __RAND_MAX
+        #pragma GCC diagnostic ignored "-Wshadow"
+        #pragma GCC diagnostic ignored "-Wfloat-equal"
```

Comment:
wow... if eigen is shadowing or use float-equal, the problem is not in PX4, but rather in eigen itself.

And you should use `#pragma GCC diagnostic push` and `#pragma GCC diagnostic pop`, otherwise you will have undesired effects.